CHANGES.md

```diff
@@ -24,7 +24,293 @@ JSDoc types are parsed in normal type annotation position but show a grammar err
+
+Corsa ignores `@class` and `@constructor`.
+This makes a difference on a function without this-property assignments or associated prototype-function assignments.
+
+6. `@param` tags now apply to at most one function.
+
+If they're in a place where they could apply to multiple functions, they apply only to the first one.
+If you have `"strict": true`, you will see a noImplicitAny error on the now-untyped parameters.
+
+```js
+/** @param {number} x */
+var f = x => x, g = x => x;
+```
+
+7. Optional marking on parameter names now makes the parameter both optional and undefined:
+
+```js
+/** @param {number} [x] */
+function f(x) {
+    return x;
+}
+```
+
+This behaves the same as Typescript's `x?: number` syntax. 
+Strada makes the parameter optional but does not add `undefined` to the type.
+
+8. Type assertions with `@type` tags now prevent narrowing of the type.
+
+```js
+/** @param {C | undefined} cu */
+function f(cu) {
+    if (/** @type {any} */ (cu).undeclaredProperty) {
+        cu // still has type C | undefined
+    }
+}
+```
+
+In Strada, `cu` incorrectly narrows to `C` inside the `if` block, unlike with TS assertion syntax.
+In Corsa, the behaviour is the same between TS and JS.
+
+### Expandos
+
+1. Expando assignments of `void 0` are no longer ignored as a special case:
+
+```js
+var o = {}
+o.y = void 0
+```
+
+creates a property `y: undefined` on `o` (which will widen to `y: any` if strictNullChecks is off).
+
+2. A this-property expression with a type annotation in the constructor no longer creates a property:
+
+```js
+class SharedClass {
+    constructor() {
+        /** @type {SharedId} */
+        this.id;
+    }
+}
+```
+
+Provide an initializer or use a property declaration in the class body:
+
+```js
+class SharedClass1 {
+    /** @type {SharedId} */
+    id;
+}
+class SharedClass2 {
+    constructor() {
+        /** @type {SharedId} */
+        this.id = 1;
+    }
+}
+```
+
+3. Assigning an object literal to the `prototype` property of a function no longer makes it a constructor function:
+
+```js
+function Foo() {}
+Foo.prototype = {
+    /** @param {number} x */
+    bar(x) {
+        return x;
+    }
+};
+```
+
+If you still need to use constructor functions instead of classes, you should declare methods individually on the prototype:
+
+```js
+function Foo() {}
+/** @param {number} x */ 
+Foo.prototype.bar = function(x) {
+    return x;
+};
+```
+
+Although classes are a much better way to write this code.
+
+### CommonJS
+
+1. Chained exports no longer work:
+
+```js
+exports.x = exports.y = 12
+```
+
+Now only exports `x`, not `y` as well.
+
+2. Exporting `void 0` is no longer ignored as a special case:
+
+```js
+exports.x = void 0
+// several lines later...
+exports.x = theRealExport
+```
+
+This exports `x: undefined` not `x: typeof theRealExport`.
+
+3. Type info for `module` shows a property with name of the export instead of `exports`:
```

Comment:
I'm not sure what this one means. Are you talking about hover? or something else?

`module`, and `exports` are different symbols entirely, right?

Reply:
It's for hover, yes. module and exports are different symbols.

Reply:
Probably this is one we can avoid mentioning for end users. But the doc doesn't structure it like that of course so it's okay.